content/en/blog/2022/etcd-integrates-continuous-fuzzing.md

```diff
@@ -0,0 +1,31 @@
+
+## What is fuzzing?
+
+Fuzzing is a technique used to automate parts of the software testing process by way of a form of stress testing. The key idea is to write a fuzzing harness, which is similar in nature to a unit- or integration test, that will execute the application under test with some arbitrary input. The fuzzing engine that will run the fuzzing harness then uses genetic algorithms to extrapolate inputs that will cause the code under test to execute in a unique manner, i.e. generate inputs that trigger new code execution paths. The goal is then to observe if the code under test misbehaves in the event of any of the generated inputs. Fuzzing has been effective in uncovering reliability bugs and vulnerabilities in software for more than two decades, and open source software is increasingly adopting the technique. 
+
+## Etcd fuzzing overview
+
+In this engagement the goal was to write a set of fuzzers that would cover a lot of the etcd codebase and also integrate the setup into the open source fuzzing service OSS-Fuzz. OSS-Fuzz is a free service offered by Google for critical open source projects to run their fuzzers continuously and report any crashes. Continuous analysis is important due to fuzzing relying on genetic algorithms, which effectively means the fuzzers will improve over time, and OSS-Fuzz will run the fuzzers on a daily basis indefinitely. In addition to this, continuous analysis is crucial for capturing any regressions that may occur.
+
+Etcd is written in the Go programming language which makes it safe from memory-corruptions. Fuzzing Go will result in finding panics such as slice/index out of range, nil-pointer dereferences, invalid type assertions, timeouts, out of memory. At the end of this engagement a total of 8 issues were found of which all were fixed. They are broken down as such:
```

Comment:
```suggestion
Etcd is written in the Go programming language which makes it safe from memory-corruptions. Fuzzing Go will find panics such as slice/index out of range, nil-pointer dereferences, invalid type assertions, timeouts, out of memory. At the end of this engagement, eight issues were found, all of which were fixed. They are broken down as such: Fuzzing Go will result in finding panics such as slice/index out of range, nil-pointer dereferences, invalid type assertions, timeouts, out of memory. At the end of this engagement a total of 8 issues were found of which all were fixed. They are broken down as such:
```

Reply:
It looks like something should have been removed here.

Reply:
There are now two "Fuzzing Go will..." sentences.

Reply:
oops, sorry, copy paste error

Reply:
I've fixed the suggestion.

Reply:
thanks!